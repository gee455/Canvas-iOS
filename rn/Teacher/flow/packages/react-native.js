//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
