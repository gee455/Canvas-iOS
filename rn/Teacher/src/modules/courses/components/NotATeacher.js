//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/**
 * @flow
 */

import React, { PureComponent } from 'react'
import {
  View,
  StyleSheet,
  Button,
  Image,
  TouchableOpacity,
  NativeModules,
  Linking,
} from 'react-native'
import i18n from 'format-message'
import { Text, Paragraph } from '../../../common/text'
import Images from '../../../images'
import Screen from '../../../routing/Screen'

export class NotATeacher extends PureComponent<{}> {
  logout = () => {
    NativeModules.NativeLogin.logout()
  }

  openStudent = () => {
    Linking.openURL('https://itunes.apple.com/us/app/canvas-by-instructure/id480883488?ls=1&mt=8')
  }

  openParent = () => {
    Linking.openURL('https://itunes.apple.com/us/app/canvas-parent/id1097996698?ls=1&mt=8')
  }

  render () {
    let title = i18n('Not a teacher?')
    let subtitle = i18n('One of our other apps might be a better fit. Tap one to visit the App Store.')
    let buttonText = i18n('Try logging in again')

    return (
      <Screen navBarHidden>
        <View style={style.container}>
          <View style={style.subContainer}>
            <Text style={{ fontSize: 30 }} testID='no-teacher.title'>{title}</Text>
            <Paragraph style={style.paragraph} testID='no-teacher.title'>{subtitle}</Paragraph>
          </View>
          <View style={style.subContainer}>
            <TouchableOpacity onPress={this.openStudent}
              testID='no-teacher.open-student'
              accessibilityTraits="button"
              accessibilityLabel={i18n('Canvas Student App')}>
              <Image source={Images.noTeacher.student} />
            </TouchableOpacity>
            <TouchableOpacity style={{ marginTop: 36 }}
              onPress={this.openParent} testID='no-teacher.open-parent'
              accessibilityTraits="button"
              accessibilityLabel={i18n('Canvas Parent App')}>
              <Image source={Images.noTeacher.parent} />
            </TouchableOpacity>
          </View>
          <View style={style.subContainer}>
            <Button title={buttonText} testID='no-teacher.logout' onPress={this.logout} />
          </View>
        </View>
      </Screen>
    )
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
  subContainer: {
    alignItems: 'center',
    margin: 8,
  },
  paragraph: {
    fontSize: 16,
    textAlign: 'center',
    padding: 8,
    maxWidth: 320,
  },
})

export default (NotATeacher: any)
