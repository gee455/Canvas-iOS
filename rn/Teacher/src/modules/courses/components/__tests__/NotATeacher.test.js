//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/**
 * @flow
 */

import { NativeModules, Linking } from 'react-native'
import React from 'react'
import NotATeacher from '../NotATeacher'
import explore from '../../../../../test/helpers/explore'

jest.mock('TouchableOpacity', () => 'TouchableOpacity')

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders NotATeacher correctly', () => {
  let tree = renderer.create(
    <NotATeacher />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('all the tapping should work', () => {
  let tree = renderer.create(
    <NotATeacher />
  ).toJSON()

  const logout = explore(tree).selectByID('no-teacher.logout') || {}
  const student = explore(tree).selectByID('no-teacher.open-student') || {}
  const parent = explore(tree).selectByID('no-teacher.open-parent') || {}

  logout.props.onPress()
  expect(NativeModules.NativeLogin.logout).toHaveBeenCalled()
  student.props.onPress()
  expect(Linking.openURL).toHaveBeenCalledWith('https://itunes.apple.com/us/app/canvas-by-instructure/id480883488?ls=1&mt=8')
  parent.props.onPress()
  expect(Linking.openURL).toHaveBeenCalledWith('https://itunes.apple.com/us/app/canvas-parent/id1097996698?ls=1&mt=8')
})
