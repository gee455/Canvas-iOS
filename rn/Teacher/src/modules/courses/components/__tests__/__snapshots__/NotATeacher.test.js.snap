// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotATeacher correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "margin": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 30,
            },
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
        testID="no-teacher.title"
      >
        Not a teacher?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#7F91A7",
              "fontSize": 16,
              "lineHeight": 23,
            },
            Object {
              "fontSize": 16,
              "maxWidth": 320,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
        testID="no-teacher.title"
      >
        One of our other apps might be a better fit. Tap one to visit the App Store.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "margin": 8,
        }
      }
    >
      <TouchableOpacity
        accessibilityLabel="Canvas Student App"
        accessibilityTraits="button"
        onPress={[Function]}
        testID="no-teacher.open-student"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/Canvas-Student.png",
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Canvas Parent App"
        accessibilityTraits="button"
        onPress={[Function]}
        style={
          Object {
            "marginTop": 36,
          }
        }
        testID="no-teacher.open-parent"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/Canvas-Parent.png",
            }
          }
        />
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "margin": 8,
        }
      }
    >
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        disabled={undefined}
        hasTVPreferredFocus={undefined}
        onPress={[Function]}
        testID="no-teacher.logout"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Try logging in again
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</RCTSafeAreaView>
`;
